Develop

Shaniser opened the pull request:
Может уже в мастер?

whitequartz committed:
Добавлен персонаж

whitequartz committed:
Update images

whitequartz committed:
Update from master

whitequartz committed:
Update engine.js

whitequartz committed:
Update from master

Shaniser committed:
Fix

Mitbus committed:
basic mouse

Mitbus committed:
Merge branch 'Player' of github.com:bmstu-iu9/utp2019-2-2d-game into Player

Mitbus committed:
added controller mouse

Shaniser committed:
mouseControl1

Shaniser committed:
mc2

Mitbus committed:
check world

Shaniser committed:
mc3

Shaniser committed:
mc4

Пересчёт теней

Mitbus committed:
middle pos player

whitequartz committed:
Отключено контекстное меню на ПКМ

whitequartz committed:
Merge branch 'Player' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Player

Mitbus committed:
mouse left click added

Mitbus committed:
min radius

Mitbus committed:
place block

Mitbus committed:
destroy timer

Shaniser committed:
cm6

Размещение блоков и освещение(не доделано)

Mitbus committed:
fix update block

Mitbus committed:
fix

Shaniser committed:
mc7

Исправлено освещение при установке блоков

Shaniser committed:
Merge branch 'Player' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Player

Shaniser committed:
cm8

Shaniser committed:
Исправлено освещение в воде

В blockTable добавлен параметр isNaturalLight

Shaniser committed:
Исправлен искусственный свет

Shaniser committed:
Плавное движение камеры

Mitbus committed:
Revert "fix"

This reverts commit 70bd6d068ab078ec33f664e13652d40f1faf53d9.

Mitbus committed:
fix controller

Mitbus committed:
add x normal

Mitbus committed:
back

Mitbus committed:
merge prepare

Mitbus committed:
Merge branch 'develop' into Player

whitequartz committed:
Тестовое исправление дрожания

whitequartz committed:
Update engine.js

Shaniser committed:
Возвращено неподвижное состояние камеры

Irautak committed:
Добавлено поведение воды ( пока без суммирования наполнености) и немного изменено поведение листвы (см. комментарий в коде)

Shaniser committed:
Изменена логика строительства

Исправлен баг с онднообразной генерацией мира

Shaniser committed:
Можна ставить блоки рядом с неосязаемыми

Mitbus committed:
Merge pull request #70 from bmstu-iu9/Player

Player

Mitbus committed:
плавное движение

Shaniser committed:
Возможно улучшено движение камеры

Shaniser committed:
Merge branch 'Player' into develop

Shaniser committed:
Можно ломать блоки за водой

Добавлена максимальная скорость падения игрока

Shaniser committed:
Исправлена возможность приставлять блоки к воде

Mitbus committed:
deltaTime fix

Размерность deltaTime теперь [сек]

Shaniser committed:
Исправлен баг с пролетанием через блоки при низком фпс

Shaniser committed:
Merge branch 'Player' into develop

Mitbus committed:
Demo save

Сохранение небольшого мира

Irautak committed:
Фикс поведения воды в водоемах + добавление понятия "твердости блока" (см. описание в начале файла Gamearea)

Кроме того, у воды была убрана гравитация, т.к она не очень совмещается
с разлитием в стиле Minecraft'а

Irautak committed:
Небольшое изменение поведения

Irautak committed:
fix условия

Irautak committed:
Добавление твердости в таблицу (см. описание в файле gamearea) + убрана гравитация у воды

Shaniser committed:
Inventory added

Shaniser committed:
Исправлена установка блоков в игрока

Shaniser committed:
Добавлено движение в воде

Shaniser committed:
Откалибрована плотность

Рекомендумая плотность:
Вода - 0.5
Лава - 0.9

Shaniser committed:
Элементы здоровья и дыхания

Добавлена работа со здоровьем:
 * Получение урона
 * Лечение
 * Смерть
 * Респаун
 * Урон от падения
 * Урон от удушья

Добавлено дыхание

Shaniser committed:
FIxes

Shaniser committed:
Merge branch 'Player' into develop

Shaniser committed:
Fixed deltaTime

Shaniser committed:
Error fixed

VolkovNik committed:
Create items.js

Mitbus committed:
Merge branch 'IlyaKarkin/issue-77' into develop

Mitbus committed:
Merge pull request #81 from bmstu-iu9/develop

Develop

Mitbus committed:
Работа с одним сохранением

Можно создавать 1 сохранение, проблема с возвратом из загрузки

VolkovNik committed:
Исправлены ошибки, добавлено железо

Добавил пропущенные запятые, теперь из травы всегда дропается dirt

VolkovNik committed:
IsGoodMelting изменено на MeltingId

VolkovNik committed:
опечатка

VolkovNik committed:
изменены имена

теперь поле name хранит в себе имя, которое мы хотим видеть в игре в инвентаре

VolkovNik committed:
Update items.js

VolkovNik committed:
добавлено железо и золото

VolkovNik committed:
fix запятые

whitequartz committed:
Откат к более старой версии + подготовка к переходу к 3 версии + небольшой рефакторинг

Mitbus committed:
Save&load added

Добавлена работа с одиночными сохранениями

Mitbus committed:
fix func

Shaniser committed:
blockTable.js полностю заменён на items.js

Shaniser committed:
Merge branch 'develop' into items-and-crafts

whitequartz committed:
Merge from develop

whitequartz committed:
Rename WorldGeneration.js to worldGeneration.js

whitequartz committed:
Rename GameArea.js to gameArea.js

Mitbus committed:
Merge pull request #84 from bmstu-iu9/items-and-crafts

Items

Mitbus committed:
Merge pull request #85 from bmstu-iu9/develop

Develop

Mitbus committed:
fixed loading

Добавлено быстрое сохранение на СКМ

Mitbus committed:
timeOdDay fix

whitequartz committed:
Added overflow

whitequartz committed:
Merge branch 'Whitequartze-path' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Whitequartze-path

whitequartz committed:
Добавлен фрембуфер. Работа движка пока нестабильна

Shaniser committed:
bug fixed, blockTable deleted

Исправлена ошибка, когда блоком в руке моментально разрушались другие блоки этого типа

Shaniser committed:
Исправлен баг спавна игрока в блоке

Переменные playerFloatX, Y заменены на player.fx, y соотвественно

Shaniser committed:
fixes

Mitbus committed:
Merge branch 'develop' into IlyaKarkin/issue-77

Mitbus committed:
Merge pull request #88 from bmstu-iu9/IlyaKarkin/issue-77

Ilya karkin/issue 77

Shaniser committed:
background added to items

Shaniser committed:
back

whitequartz committed:
Переход к 3 версии движка

Mitbus committed:
added load

Mitbus committed:
load player fx\fy

whitequartz committed:
Added new render

Shaniser committed:
Refactor items.js

Mitbus committed:
fixed

Mitbus committed:
Merge branch 'Test' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Test

Mitbus committed:
fixmerge

whitequartz committed:
Улучшение производительности

whitequartz committed:
Удалён мусор

whitequartz committed:
Полноценная работа с фреймбуфером

Очень сильно улучшена производительность
Добавлены новые баги)

Shaniser committed:
Добавлена работа с задним слоем

Для взаимодействия кнопка shift

Shaniser committed:
Some console.logs deleted

Shaniser committed:
Merge branch 'develop' into Player

Shaniser committed:
Добавлено замедленное передвижение с шифтом

Исправлен баг с застреванием в углу блоков

whitequartz committed:
Добавлен фон

Shaniser committed:
Добавлены озера с лавой

whitequartz committed:
Обновлён новый рендер

whitequartz committed:
Small fixes

Mazdaywik commented on Scripts/gameArea.js:
```diff
@@ -39,7 +39,7 @@ class GameArea{
         // Возвращает освещение конкретного блока
         this.getLight = (x, y) => {
-            let grad = (y > this.elevationMap[x]) ? 1 : ((y < 0.7 * this.elevationMap[x]) ? 0.2 : ((y - 0.7 * this.elevationMap[x]) / (0.3 * this.elevationMap[x]) * 0.8 + 0.2));
+            let grad = (y > this.elevationMap[x]) ? 1 : ((y < 0.9 * this.elevationMap[x]) ? 0.2 : ((y - 0.9 * this.elevationMap[x]) / (0.1 * this.elevationMap[x]) * 0.8 + 0.2));
```
Я насчитал 178 символов, что на 25 % больше лимита в Wiki (120 символов).

Причём оператор `? :` очень хорошо разбивается на строчки:
```JavaScript
            let grad =
                (y > this.elevationMap[x]) ? 1 :
                ((y < 0.7 * this.elevationMap[x]) ? 0.2 :
                ((y - 0.7 * this.elevationMap[x]) / (0.3 * this.elevationMap[x]) * 0.8 + 0.2));
```

Mazdaywik reviewed (changes requested):
Есть нарушение стиля оформления кода. Содержательно смысл кода проверить не могу, поскольку в ваших исходниках не разбираюсь.

Shaniser committed:
constant fix

whitequartz committed:
Merge branch 'develop' into Graphical-Engine

whitequartz committed:
Delete files

Mitbus committed:
style fix

Shaniser committed:
Merge branch 'develop' into Player

whitequartz committed:
Перемезение шейдеров в отдельный файл

Mitbus committed:
style fix 2

Исправлено нарушение стился в некотрых файлах, возможно, есть еще нарушения. Также убрал ненужный код в gameArea.js

Mitbus commented:
В блоке поведения воды в `gameArea.js` некоторые условия слишком сложные, еще и оформлены так, что их трудно читать.

whitequartz commented:
> В блоке поведения воды в `gameArea.js` некоторые условия слишком сложные, еще и оформлены так, что их трудно читать.

@Irautak

Mitbus committed:
style fix 3

Mitbus commented:
Посмотрел все, кроме `gameArea.js`, `player.js`, `worldGeneration.js`, `level.js`. В них не могу проверить смысл кода т.к. знаю новые места в них слишком поверхностно.

whitequartz committed:
Rename scale into blockSize

whitequartz committed:
Исправление щелей между чанками

whitequartz committed:
Доделан метод createChunk

Mitbus committed:
fix points #86

Исправле баг с точками на фоне

Mazdaywik commented:
> В блоке поведения воды в `gameArea.js` некоторые условия слишком сложные, еще и оформлены так, что их трудно читать.

Хороший способ борьбы со сложными условиями — вынесение их в отдельную функцию (метод) с понятным именем. Либо используя вспомогательные булевские переменные, но тоже с понятными именами.

Например:
```JavaScript
                if (n > 0
                        && ((startX - x) * (startX - x) + (startY - y) * (startY - y)
                            < (startX - nextX) * (startX - nextX) + (startY - nextY) * (startY - nextY))
                        && nextX >= 0
                        && nextY >= 0
                        && nextX < width
                        && nextY < height
                        && (shadowMap[nextX][nextY] === undefined
                            || (isNatural && shadowMap[nextX][nextY] % 1000 < n)
                            || (!isNatural && Math.floor(shadowMap[nextX][nextY] / 1000) < n))) {
```
Это можно упростить так:
```JavaScript
                const distStart = (x, y) => ((startX - x) * (startX - x) + (startY - y) * (startY - y));
                const inRange = (min, val, max) = ((min <= val) && (val < max));
                if (n > 0
                        && (distStart(x, y) < distStart(nextX, nextY))
                        && inRange(0, nextX, width)
                        && inRange(0, nextY, height)
                        && (shadowMap[nextX][nextY] === undefined
                            || (isNatural && shadowMap[nextX][nextY] % 1000 < n)
                            || (!isNatural && Math.floor(shadowMap[nextX][nextY] / 1000) < n))) {
```
Можно упрощать и дальше:
```JavaScript
                const distStart = (x, y) => ((startX - x) * (startX - x) + (startY - y) * (startY - y));
                const inRange = (min, val, max) = ((min <= val) && (val < max));
                const nextIsFurther = distStart(x, y) < distStart(nextX, nextY);
                const nextInScreen = inRange(0, nextX, width) && inRange(0, nextY, height);
                if (n > 0
                        && nextIsFurther
                        && nextInScreen
                        && (shadowMap[nextX][nextY] === undefined
                            || (isNatural && shadowMap[nextX][nextY] % 1000 < n)
                            || (!isNatural && Math.floor(shadowMap[nextX][nextY] / 1000) < n))) {
```
И ещё дальше:
```JavaScript
                const distStart = (x, y) => ((startX - x) * (startX - x) + (startY - y) * (startY - y));
                const inRange = (min, val, max) = ((min <= val) && (val < max));
                const nextIsFurther = distStart(x, y) < distStart(nextX, nextY);
                const nextInScreen = inRange(0, nextX, width) && inRange(0, nextY, height);
                const acceptableWTFofShadowMap = shadowMap[nextX][nextY] === undefined
                    || (isNatural && shadowMap[nextX][nextY] % 1000 < n)
                    || (!isNatural && Math.floor(shadowMap[nextX][nextY] / 1000) < n)
                if (n > 0 && nextIsFurther && nextInScreen && acceptableWTFofShadowMap) {
```
Упростил так, что выражение под `if` влезло в одну строчку.

Смысл последнего условия про `shadowMap` от меня далёк, надеюсь, вы сами сможете упростить код так, чтобы даже я его понял.

Вообще, смысл условия мне не ясен, поэтому я разбирал его наугад. Скорее всего, я где-то ошибся. Я думаю, что автор кода, расписывая условие подобным образом, всё всем (включая меня) станет понятнее.

Mazdaywik reviewed (dismissed):
Посмотрите внимательно Files changed. Зачем нужно делать исполнимые картинки? Исполнимые! Картинки! Зачем 755 для `player.png`?

Чтобы атрибуты сделать корректными (и логичными), нужно выполнить следующие команды. На Windows;
```Batch
git update-index --chmod=-x Images/*.png LICENCE README.md (и другие файлы)
git commit -m "Исправлены атрибуты у некоторых файлов
```
На unix-like:
```Bash
chmod -x Images/*.png LICENCE README.md (и другие файлы)
git add -p
(на запросы изменения флагов отвечаем y, на остальные n)
git commit -m "Исправлены атрибуты у некоторых файлов"
```

(Да, кавычку на Windows можно не закрывать)

Mazdaywik commented:
Вы, кстати, распишите словами, что означает это длинное условие, и я предложу более точный вариант его декомпозиции.

Mazdaywik commented on Scripts/gameArea.js:
```diff
@@ -267,28 +267,42 @@ class GameArea{
+                let startX = Math.floor(player.x - Player.WIDTH / 2);
+                let endX = Math.floor(player.x + Player.WIDTH / 2);
+                let startY = Math.floor(player.y);
+                let endY = Math.floor(player.y + Player.HEIGHT);
```
Хороший стиль, который пропагандировал @alex-qrcs, писать `const` вместо `let` везде, где только можно. Здесь, например, лучше написать 4 `const`’а.

Mitbus committed:
style fix

whitequartz committed:
Исправлены атрибуты у некоторых файлов

whitequartz committed:
Удаление для исправления

whitequartz committed:
Удаление для исправления

whitequartz committed:
Удаление для исправления

whitequartz committed:
Исправление аттрибутов файлов

Mitbus commented on Scripts/gameArea.js:
```diff
@@ -145,145 +156,254 @@ class GameArea{
+                        && this.map[x][y + 1][layout] !== 8
+                        && (!this.map[x + 1][y][layout]
+                            || (this.map[x + 1][y][layout] - 9000) % 8 >= power
+                            ||  this.map[x + 1][y][layout] < 9000)
```
Мне кажется, что помимо вынесения однотипных условий в функцию стоит еще пояснить, что за константы используются и почему именно они. Много где в коде такое встречается

Mitbus commented on Scripts/gameArea.js:
```diff
@@ -145,145 +156,254 @@ class GameArea{
+                            }, 50);
+                    } else {
+
+                        let direction = Math.floor((+block.id - 9000) / 8);
```
Почему мы храним id блока строкой, но при каждом обращении переводим его к числу?

Mitbus commented on Scripts/level.js:
```diff
@@ -1,107 +1,335 @@
+    					let effK = ((player.hand.item && player.hand.info.isTool
+    						&& currentBlock.type == player.hand.info.type))
+    					? player.hand.info.efficiency : 1;
+    					currentBlock.durability -= deltaTime * effK;
```
Точно не знаю, в чем разница, но слышал рекомендацию использовать `===` вместо `==`. И еще тут нарушена стилизация кода

whitequartz committed:
Теперь при загрузке голубой экран

Irautak committed:
Починил деревья и лут с них

Irautak committed:
Добавлено (забытое) условие на проверку выхода за карту

Irautak committed:
Адаптация поведения воды для различных жидкостей

Лава разливается как вода

whitequartz committed:
Удаление для исправление атрибутов

whitequartz committed:
Исправление атрибутов файлов

Mitbus commented:
@Irautak, почему ты в `gameArea.js` используешь стрелочные функции? Насколько я понимаю, в этом случае лучше использовать `function`. Если ты пишешь `this.a = () => {...};` , то ты фактически для каждого объекта класса объявляешь поле и кладешь туда функцию, к тому же из этой функции недоступен `this`. В данном случае мы создаем не более одного объекта класса, поэтому нет почти никаких проблем, но в общем случае нужно быть внимательным.

whitequartz commented:
> @Irautak, почему ты в `gameArea.js` используешь стрелочные функции? Насколько я понимаю, в этом случае лучше использовать `function`. Если ты пишешь `this.a = () => {...};` , то ты фактически для каждого объекта класса объявляешь поле и кладешь туда функцию, к тому же из этой функции недоступен `this`. В данном случае мы создаем не более одного объекта класса, поэтому нет почти никаких проблем, но в общем случае нужно быть внимательным.

~~Нам ведь на лекциях советовали не использовать устаревший ```function```, а только анонимные функции.
Разве нельзя из анонимной функции получить ```this```?~~

Mitbus committed:
Изменено управление сохранением

f - сохраниться
g -удалить

whitequartz committed:
Исправлен небольшой баг

Исправлен баг с некорректным размером фона при изменении scale

whitequartz committed:
Добавлена надпись "Идёт загрузка игры..."

whitequartz committed:
Merge from develop

Mitbus committed:
Отслеживание редактирования

Добавлен метод, который нужно вызывать вместо присваивания для сохранения изменений

Mitbus committed:
Merge branch 'Graphical-Engine' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Graphical-Engine

Mitbus committed:
back merge

whitequartz committed:
Добавлена надпись при загрузке игры

Mazdaywik commented:
@whitequartz, а что, консольными командами исправить атрибуты не получилось? Потому что удалять и снова добавлять файл вообще говоря не очень хорошо. Гит в этом случае уже не сможет отследить историю строчек файла для команды `git blame`. Для бинарных файлов это не актуально, а вот для `README.md` Вы историю потеряли.

Mitbus committed:
added createChunk

whitequartz committed:
Create drawChunk

whitequartz committed:
Merge branch 'Graphical-Engine' of https://github.com/bmstu-iu9/utp2019-2-2d-game into Graphical-Engine

Mitbus commented:
> @whitequartz, а что, консольными командами исправить атрибуты не получилось? Потому что удалять и снова добавлять файл вообще говоря не очень хорошо. Гит в этом случае уже не сможет отследить историю строчек файла для команды `git blame`. Для бинарных файлов это не актуально, а вот для `README.md` Вы историю потеряли.

Иногда замечаю в коммитах такие изменения файлов: все содержимое удалено, а потом добавлено тоже самое. Это считается изменением истории или мы тоже ее так теряем? И еще, мы пытались переименовать файл, но гит не хотел видеть эти изменения и мы его пересоздали.

whitequartz commented:
> @whitequartz, а что, консольными командами исправить атрибуты не получилось? Потому что удалять и снова добавлять файл вообще говоря не очень хорошо. Гит в этом случае уже не сможет отследить историю строчек файла для команды `git blame`. Для бинарных файлов это не актуально, а вот для `README.md` Вы историю потеряли.

Через сайт вся история отображается
![alt text](https://i.imgur.com/1xQIIEd.jpg)

Mitbus commented:
`game.js` -> `preprocessing.js`

Mitbus commented:
Например, в этом mr `level.js` отмечен как полностью удаленный, а потом добавлен (хотя на самом деле добавлены новые методы)

Mitbus committed:
load condition

Mazdaywik commented:
@whitequartz, через командную строку нет. Вот вывод `git blame README.md`:
```
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  1) # utp2019-2-2d-game
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  2) 2D-игра, упрощённый вариант Terraria, команда Алексея Костюченко
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  3) 
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  4) Участники:
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  5) * Алексей Костюченко
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  6) * Илья Каркин
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  7) * Илья Щербаков
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  8) * Искандер Омуров
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300  9) * Никита Волков
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300 10) * Светлана Пирожкова
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300 11) * Надим Абдул Гани
cda7a636 (whitequartz 2019-07-19 16:56:35 +0300 12) * Владислав Ломтев
```

Mazdaywik commented:
@Mitbus, ГитХаб здесь показывает разницу между ветками `master` и `develop`. В рамках текущей ветки действительно были добавлены только методы. Но в ветку `master` @Shaniser закоммитил файл `level.js` с концами строк Windows (`\r\n`): 66215a6397896aa3f5f8667f84606b09adfb9ab9.

Исправил в<span>&nbsp;</span>#92. Хотя мог бы, пользуясь правом администратора, напрямую закоммитить в `master`.

Mazdaywik committed:
Исправлены unix-атрибуты для исходных файлов

Ни для одного из файлов репозитория в данной ветке атрибут исполнимости
не имеет смысла.

Mazdaywik commented:
Чтобы вы не мучались с удалением файлов из web-интерфейса, я исправил атрибуты сам: #93.

Mazdaywik committed:
Merge pull request #93 from bmstu-iu9/mazdaywik-chmod

Исправлены unix-атрибуты для исходных файлов

Shaniser committed:
Refactor my part of GameArea

Shaniser committed:
Function name fixed

Mazdaywik commented on Scripts/gameArea.js:
```diff
@@ -54,81 +54,97 @@ class GameArea{
+                            //............................................. Натуральный свет меньше n
+                            || (isNatural && this.getNaturalLight(nextX, nextY) < n)
+                            //............................................ Искуственный свет меньше n
+                            || (!isNatural && this.getArtificalLight(nextX, nextY) < n))) {
```
Во! Теперь всё понятнее. Даже без комментариев.

Можно пойти и дальше:
```JavaScript
const getLight = isNatural ? (x, y) => this.getNaturalLigth(x, y) : (x, y) => this.getArtificalLight(x, y);
if (… || (getLight(x, y) < n) {
```
Проверка `(isNatural && this.getNaturalLight(x, y) < n) || (!isNatural && this.getArtificalLight(x, y) < n))` в коде много раз повторяется, поэтому имеет смысл вынести в отдельную функцию.

Mazdaywik replied:
Добавлю. Свет он не «натуральный», а «естественный».

И ещё, комментарии не нужны. Вернее, не нужны те комментарии, которые повторяют код. Например:
```JavaScript
x += 5;  // прибавили пять

//............................................ Искуственный свет меньше n
|| (!isNatural && this.getArtificalLight(nextX, nextY) < n))) {
```

Shaniser replied:
<img width="1318" alt="Снимок экрана 2019-07-20 в 12 01 33" src="https://user-images.githubusercontent.com/48556605/61576732-27e48b00-aae6-11e9-9c07-dfb9b5f2648f.png">
Или вопрос к какому-то комменту?
UPD: вижу

Shaniser replied:
> Проверка `(isNatural && this.getNaturalLight(x, y) < n) || (!isNatural && this.getArtificalLight(x, y) < n))` в коде много раз повторяется, поэтому имеет смысл вынести в отдельную функцию.

1) isNatural локальная переменная конкретных функций
2) this.getLight уже имеется, и означает итоговую освещённость блока (введение подобной функции может привести к путанице)

Mazdaywik replied:
Я имел ввиду добавить
```JavaScript
const ligthBelowThreshold = (x, y) => (isNatural && this.getNaturalLigth(x, y) < n)
    || (!isNatural && this.getArtificalLight(x, y) < n);
```
в начале метода `addLightRound`, поскольку этот код громоздкий и повторяется дважды. А вместо `(x && y) || (!x && z)` можно записать `x ? y : z`.

Mazdaywik reviewed (commented):
Стало существенно понятнее! Хороший, годный рефакторинг!

Shaniser committed:
Refactor

Shaniser committed:
Merge branch 'develop' into Player

Shaniser committed:
Продвинутая система определения доступности блоков

Mazdaywik commented on Scripts/player.js:
```diff
@@ -74,11 +74,130 @@ class Player {
+        // Рассчитать угол до объекта
+        this.angle = (x, y) => {
+            return Math.sign(y - this.y - Player.HEIGHT / 2)
+                    * Math.acos((x - this.x) / hypotenuse(x - this.x, y - this.y - Player.HEIGHT / 2));
```
Навскидку: почему тут не используется более логичная функция [`Math.atan2`](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/atan2)?

Shaniser replied:
Деление на 0

Mazdaywik replied:
Функция `atan2` не делит на ноль. Её результат не определён для вызова `atan2(0, 0)`, но для остальных комбинаций результат осмыслен:
```JavaScript
atan2(100500, 0) = 0
atan2(0, 42) = π/2
atan2(-666, 0) = π
atan2(0, -13) = -π/2
```

Посмотрел в справку. Оказывается, у неё определён результат и для `atan2(0, 0)`.

Shaniser replied:
Оно того не стоит, в общем (на вскидку работать не начало, а у меня на реализацию всего этого часов 5 ушло)

whitequartz committed:
Добавлен значок загрузки

Irautak committed:
Добавление отслеживания изменений при установке блоков

Добавление функции @Mitbus и @whitequartz для отслеживания изменений при
установке блоков

whitequartz committed:
Merge branch 'master' into Graphical-Engine

whitequartz committed:
fix bug

whitequartz committed:
Fix bug

whitequartz committed:
Small Fixes

Mitbus committed:
drow condition

Mitbus committed:
solid field

Mitbus committed:
fix

Mitbus committed:
rename settings field

whitequartz committed:
Релиз 3 версии движка

whitequartz committed:
Fix background

whitequartz committed:
Merge branch 'develop' into Graphical-Engine

whitequartz committed:
Исправление с белыми пикселями

whitequartz committed:
Исправление с ломанием блоков

whitequartz commented:
Добавлена 3 версия графического движка

Mazdaywik commented:
Опять исполнимые CSS-ки.

Mazdaywik commented on Scripts/preprocessing.js:
```diff
@@ -125,28 +140,51 @@ image.onload = () => {
 				}
-
-				render.render(cameraX, cameraY, player.x, player.y, cameraScale, arrOfChunks);
+				
```
Пробельчик.

Mazdaywik commented on Scripts/worldGeneration.js:
```diff
@@ -593,8 +647,8 @@ const generate = (width, height, seed) => {
-                }else if(worldMap[x][y][GameArea.MAIN_LAYOUT] == 8 || worldMap[x][y][GameArea.MAIN_LAYOUT] == 9){
-                    shadowRound(x, y, x, y, Math.floor(maxLight * 2 / 3), true);
+                } else if(items[worldMap[x][y][GameArea.MAIN_LAYOUT]] != undefined && items[worldMap[x][y][GameArea.MAIN_LAYOUT]].brightness > 0){
+                    shadowRound(x, y, x, y, items[worldMap[x][y][GameArea.MAIN_LAYOUT]].brightness, items[worldMap[x][y][GameArea.MAIN_LAYOUT]].isNaturalLight === true);
```
Тут точно меньше 120 колонок?

whitequartz committed:
Исправлены атрибуты у некоторых файлов

whitequartz committed:
Небольшой рефакторинг

Mazdaywik commented on Scripts/gameArea.js:
```diff
@@ -302,19 +303,21 @@ class GameArea{
+                                if (direction === 0) {
+									setTimeout(() => {
+										if (this.map[x][y][layout] === +block.id
+											&& this.map[x - 1][y][layout] === undefined) {
```
В предыдущей строчке стало 135 символов. По-моему, у вас сбилась настройка отступов в текстовом редакторе.

whitequartz committed:
Замена пробелов на табуляции

whitequartz committed:
Замена табуляций на пробелы

Mazdaywik commented on Scripts/gameArea.js:
```diff
@@ -318,15 +318,15 @@ class GameArea{
-                                        }
-                                    }, 200);
-                                }
+											this.placeBlock(x - 1, y,
```
Да, похоже, что теперь один отступ равен 8 пробелам.

Mazdaywik replied:
Сами разбирайтесь, коммитить исходники с отступами табуляцией (знак `\t`) или N пробелов. Я не буду участвовать в этом холиваре.

Mitbus replied:
В правилах оформления кода написано: "... с помощью табуляции", хотя почти все используют 4 пробела)
@whitequartz , возможно, стоит заменить это правило на "в одном файле в общем стиле", т.к. правила появились значительно позднее и у всех уже было по-своему

whitequartz replied:
@Mitbus, отредактировал данный пункт

Mazdaywik replied:
Проблема в том, что GitHub отображает табуляции как 8 пробелов, из-за чего через сайт отступы отображаются вдвое больше. Но, я думаю, это можно потерпеть.

Важно, чтобы редакторы у всех были настроены так, чтобы не меняли автоматически `\t` на N пробелов или наоборот. Либо, если меняли, то только в тех строках, которые пользователь явно менял.

Mitbus committed:
deleteWorld added

Mitbus committed:
smart savies

Теперь сохранения происходят по ключу и изменениям мира

Mitbus committed:
merge issue-77 testing

Mitbus committed:
smart save by world gen key

Mitbus committed:
smart save by world gen key

Mitbus committed:
merge fix

Mazdaywik commented on Scripts/saveWorld.js:
```diff
@@ -41,53 +41,62 @@ const deleteDatabase = () => {
+        deleteDatabase();
     }
-
+    
```
Пробельчик. Чтобы их случайно не коммитить, пользуйтесь `git add -p`.

Mitbus committed:
deleted extra space

Mitbus committed:
fix gameAreaMapSet + exrta space

Mitbus committed:
timeOfDay save

whitequartz committed:
Добавление возможности регулирования светом для фона и персонажа

Mitbus committed:
framebuffer added

Mitbus committed:
fix bug

#96 Проблемы с отображениемм при загрузке

Mitbus committed:
back

По ошибке запушил не в ту ветку

Mitbus committed:
Merge branch 'master' into develop

whitequartz committed:
Сбор статисти

ВНИМАНИЕ! Этот commit нужен для сбора некоторой статистики во время разработки данной игры.
После сбора статистики данные изменения будут удалены.
Собираться статистика будет в основном для понимания возможностей графики на разных браузерах.

whitequartz committed:
small fix

whitequartz committed:
Статистика

Mitbus committed:
Update README.md

Mitbus committed:
Update README.md

Mitbus committed:
Update README.md

Mitbus reviewed (approved):
Думаю, что уже можно добавлять. То, что можно было исправили, а в сложных моментах знаем как можно решать такие проблемы

Mitbus commented:
@Mazdaywik , если больше нет замечаний, то принимайте
